var/spack/repos/builtin/packages/kaldi/package.py

```diff
@@ -0,0 +1,91 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+from distutils.dir_util import copy_tree
+
+
+class Kaldi(Package):
+    """Kaldi is a toolkit for speech recognition written 
+    in C++ and licensed under the Apache License v2.0. 
+    Kaldi is intended for use by speech recognition researchers."""
+
+    homepage = "https://github.com/kaldi-asr/kaldi"
+    url      = "https://github.com/kaldi-asr/kaldi/archive/master.zip"
+
+    version('master', git='https://github.com/kaldi-asr/kaldi.git')
+
+    variant('shared', default=True,
+            description='build shared libraries')
+    variant('double', default=False,
+            description='build with double precision floats')
+    variant('cuda', default=False,
+            description='build with CUDA')
+
+    depends_on('blas')
+    depends_on('speex')
+    depends_on('openfst@1.6:')
+    depends_on('cuda', when='+cuda')
+    depends_on('sph2pipe', type='run')
+    depends_on('sctk', type='run')
+
+    def install(self, spec, prefix):
+        configure_args = [
+            '--threaded-math',
+            '--speex-root=' + spec['speex'].prefix,
+            '--fst-root=' + spec['openfst'].prefix,
+            '--fst-version=' + str(spec['openfst'].version)
+        ]
+    
+        if '~shared' in spec:
+            configure_args.append('--static')
+        else:
+            configure_args.append('--shared')
+
+        if '^openblas' in spec:
+            configure_args.append('--openblas-root=' + spec['blas'].prefix)
+            configure_args.append('--mathlib=OPENBLAS')
+        elif '^atlas' in spec:
+            configure_args.append('--atlas-root=' + spec['blas'].prefix)
+            configure_args.append('--mathlib=ATLAS')
+        elif '^intel-parallel-studio' in spec or '^intel-mkl' in spec:
+            configure_args.append('--mkl-root=' + spec['blas'].prefix)
+            configure_args.append('--mathlib=MKL')
+
+        if '+cuda' in spec:
+            configure_args.append('--use-cuda=yes')
+            configure_args.append('--cudatk-dir=' + spec['cuda'].prefix)
+
+        with working_dir("src"):
+            configure(*configure_args)
+            make()
+            bash = which('bash')
+            mkdir(prefix.bin)
+            mkdir(prefix.lib)
+            mkdir(prefix.include)
+            bash('-c', "find bin -type f -executable -exec cp {} %s \;" %
+                 prefix.bin)
+            copy_tree('lib', prefix.lib)
+            bash('-c', 'rsync -a --include="*/" --include="*.h" --exclude="*"\
+                ./ {0}'.format(prefix.include))
```

Comment:
Can we use Unix `install` for this?  Or an Internal Spack/Python file copy subroutine (as is used in other packages; see `googletest`).  Are there any file permission issues that must be addressed?